@Hoikas A few questions before proceeding.

----

On `x86_windows`, I get this error:

```
Found the following errors:
Error: While reading versions for port igraph from file: C:\a\1\s\versions\i-\igraph.json
       File declares version `0.9.2` with SHA: c3d0676f42963d1d217093232a21cbeb2f81d152
       But local port with the same verion has a different SHA: e5cf4a1ffbdc577ca2de2493b2eebf7e0a73bf0c
       Please update the port's version fields and then run:

           vcpkg x-add-version igraph

       to add a new version.
```

I have already done `vcpkg x-add-version igraph` and committed the result. What is wrong? And why only on a single platform?

----

On UWP platforms I get:

```
error MSB4181: The "CompileXaml" task returned false but did not log an error. 
```

I have no experience with this platform, nor a Windows machine at hand. Any quick tips to try out? If not, I will blacklist this platform.

--- a/ports/igraph/001_add_crt_secure_no_warnings.patch
+++ b/ports/igraph/001_add_crt_secure_no_warnings.patch
@@ -1,5 +1,5 @@
 diff --git a/etc/cmake/compilers.cmake b/etc/cmake/compilers.cmake
-index 3aabff4cb..766031bcd 100644
+index 3aabff4cb..97c01e545 100644
 --- a/etc/cmake/compilers.cmake
 +++ b/etc/cmake/compilers.cmake
 @@ -2,6 +2,7 @@ include(CheckCCompilerFlag)
@@ -10,12 +10,3 @@ index 3aabff4cb..766031bcd 100644
  endif()
  
  if (NOT MSVC)
-@@ -11,7 +12,7 @@ endif()
- macro(use_all_warnings TARGET_NAME)
-   if(MSVC)
-     target_compile_options(${TARGET_NAME} PRIVATE 
--      /W4 # enable most warnings, then disable:
-+      /W3 # enable most warnings, then disable:
-       /wd4244 # 'conversion' conversion from 'type1' to 'type2', possible loss of data
-       /wd4267 # 'var' : conversion from 'size_t' to 'type', possible loss of data 
-       /wd4996 # deprecated functions, e.g. 'sprintf': This function or variable may be unsafe. Consider using sprintf_s instead.

--- a/ports/igraph/arith_win32.h
+++ b/ports/igraph/arith_win32.h
@@ -3,5 +3,6 @@
 #define IEEE_8087
 #define Arith_Kind_ASL 1
 #define Double_Align
+#define NANCHECK
 #define QNaN0 0x0
 #define QNaN1 0x7ff80000

--- a/ports/igraph/portfile.cmake
+++ b/ports/igraph/portfile.cmake
@@ -37,12 +37,17 @@ vcpkg_configure_cmake(
     PREFER_NINJA
     OPTIONS
         -DIGRAPH_ENABLE_LTO=AUTO
+        # ARPACK not yet available in vcpkg.
         -DIGRAPH_USE_INTERNAL_ARPACK=ON
+        # OpenBLAS provides BLAS/LAPACK but some tests fail with OpenBLAS on Windows.
+        # See https://github.com/igraph/igraph/issues/1491
         -DIGRAPH_USE_INTERNAL_BLAS=ON
-        -DIGRAPH_USE_INTERNAL_CXSPARSE=ON
-        -DIGRAPH_USE_INTERNAL_GLPK=ON
-        -DIGRAPH_USE_INTERNAL_GMP=ON
         -DIGRAPH_USE_INTERNAL_LAPACK=ON
+        -DIGRAPH_USE_INTERNAL_CXSPARSE=OFF
+        # GLPK is not yet available in vcpkg.
+        -DIGRAPH_USE_INTERNAL_GLPK=ON
+        # Currently, external GMP provides no performance of functionality benefits.
+        -DIGRAPH_USE_INTERNAL_GMP=ON
         -DF2C_EXTERNAL_ARITH_HEADER=${ARITH_H}
         ${FEATURE_OPTIONS}
 )

--- a/ports/igraph/vcpkg.json
+++ b/ports/igraph/vcpkg.json
@@ -3,6 +3,9 @@
   "version-string": "0.9.2",
   "description": "igraph is a C library for network analysis and graph theory, with an emphasis on efficiency portability and ease of use.",
   "homepage": "https://igraph.org/",
+  "dependencies": [
+    "suitesparse"
+  ],
   "default-features": [
     "graphml"
   ],

--- a/versions/i-/igraph.json
+++ b/versions/i-/igraph.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "c3d0676f42963d1d217093232a21cbeb2f81d152",
+      "git-tree": "bd45c60c59bb433084897b194023c42423e5a77f",
       "version-string": "0.9.2",
       "port-version": 0
     }

--- a/versions/i-/igraph.json
+++ b/versions/i-/igraph.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "e5cf4a1ffbdc577ca2de2493b2eebf7e0a73bf0c",
+      "git-tree": "c3d0676f42963d1d217093232a21cbeb2f81d152",
       "version-string": "0.9.2",
       "port-version": 0
     }

--- a/versions/i-/igraph.json
+++ b/versions/i-/igraph.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "bd45c60c59bb433084897b194023c42423e5a77f",
+      "git-tree": "070fd2c624494271163192e9c6d438244b294404",
       "version-string": "0.9.2",
       "port-version": 0
     }

